src/Handler/ExceptionHandler.php

```diff
@@ -74,77 +83,116 @@ public function __invoke(
     ) {
         try {
             return $next($request, $response);
-        } catch (Exception $e) {
-            if ($this->logger) {
-                $this->logger->error($e->getMessage(), [
-                    'exception' => $e
-                ]);
-            }
-
-            $type = $this->type($request);
-
-            $response = $response->withHeader('Content-Type', $type);
-
-            try {
-                if (method_exists($e, 'getHttpStatus')) {
-                    $code = $e->getHttpStatus();
-                } else {
-                    $code = $e->getCode();
-                }
-                $response = $response->withStatus($code);
-            } catch (InvalidArgumentException $_) {
-                // Exception did not contain a valid code
-                $response = $response->withStatus(500);
-            }
-
-            if ($e instanceof HttpException) {
-                $response = $e->withResponse($response);
-            }
-
-            $handler = $this->handler($type);
-            $this->whoops->pushHandler($handler);
-
-            $body = $this->whoops->handleException($e);
-            $response->getBody()->write($body);
-
-            $this->whoops->popHandler();
-
-            return $response;
+        } catch (ExceptionInterface $exception) {
```

Comment:
I think I would rather see this implemented as an injection, where the ExceptionHandler takes a type of object that converts an exception into a Payload and then formats it using a Formatter.

The current implementation is not very flexible for user extension and doesn't allow for writing custom handlers based on 3rd party exception types.